astropy/coordinates/baseframe.py

```diff
@@ -902,16 +1137,27 @@ def __getattr__(self, attr):
             self.data  # this raises the "no data" error by design - doing it
             # this way means we don't have to replicate the error message here
 
-        else:
+        elif attr in self.representation_component_names:
             rep = self.represent_as(self.representation, in_frame_units=True)
             val = getattr(rep, self.representation_component_names[attr])
             return val
 
+        elif attr in self.differential_component_names:
+            # TODO: this doesn't work for the case when there is only
```

Comment:
Well, one more reason for getting rid of this horrible `in_frame_units`.... (sorry, I know, not helpful ;-)